mlir/lib/Catalyst/Transforms/TBAAPatterns.cpp

```diff
@@ -31,7 +31,13 @@ void setTag(mlir::Type baseType, catalyst::TBAATree *tree, mlir::MLIRContext *ct
 {
     mlir::LLVM::TBAATagAttr tag;
     if (isa<IndexType>(baseType) || isa<IntegerType>(baseType)) {
-        tag = tree->getTag("int");
+        // Index can be used as a pointer.
+        if (isa<IndexType>(baseType) && (isa<LLVM::StoreOp>(newOp) || isa<LLVM::LoadOp>(newOp))) {
```

Comment:
I don't know if we should always consider index as a pointer. Rather, I would walk backwards and if the index comes from an op that we know represents a pointer (like `memref.extract_aligned_pointer`), we change the TBAA to pointer. 
Always considering index to be a pointer sounds like it produce a lot of errors.

Reply:
I am not sure if there is any way to memerize if a memref consider from `memref.extract_aligned_pointer`.
Maybe we need to modify the mlir side to add this information?